flake.nix

```diff
@@ -0,0 +1,158 @@
+# SPDX-FileCopyrightText: 2022 Oxhead Alpha
+# SPDX-License-Identifier: MIT
+
+{
+  description = "The stablecoin flake";
+
+  nixConfig.flake-registry = "https://gitlab.com/morley-framework/morley-infra/-/raw/main/flake-registry.json";
+
+  inputs.morley-infra.url = "gitlab:morley-framework/morley-infra";
+  inputs.morley.url = "gitlab:morley-framework/morley";
+
+  outputs = { self, flake-utils, haskell-nix, morley-infra, morley, ... }:
+    (flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
+      let
+        pkgs = morley-infra.legacyPackages.${system};
+
+        ligo = pkgs.stdenv.mkDerivation {
+          name = "ligo";
+          src = pkgs.fetchurl {
+            url = "https://gitlab.com/ligolang/ligo/-/jobs/2684632179/artifacts/raw/ligo";
+            sha256 = "sha256-sr1gxTDB99F3Cmm2tl2280MAOCT4ONtW9Xv69NdOqIk=";
+          };
+          dontUnpack = true;
+          installPhase = ''
+            mkdir -p $out/bin
+            cp $src $out/bin/ligo;
+            chmod +x $out/bin/ligo
+          '';
+        };
+        ligoVersion = "0.46.1";
+
+        local-packages = [{
+            name = "stablecoin";
+            subdirectory = ".";
+        }];
+        local-packages-names = map (p: p.name) local-packages;
+
+        projectSrc = ./haskell;
+
+        cabalFile = pkgs.runCommand "stablecoin.cabal" {} ''
+          ${pkgs.haskellPackages.hpack}/bin/hpack ${projectSrc} - > $out
+        '';
+
+        hs-pkgs = pkgs.haskell-nix.stackProject {
+          # project src with .cabal file added
+          src = pkgs.runCommand "src-with-cabal" {} ''
+            cp -r --no-preserve=mode ${projectSrc} $out
+            cp ${cabalFile} $out/stablecoin.cabal
+          '';
+          ignorePackageYaml = true;
+          modules = [
+            {
+              packages = pkgs.lib.genAttrs local-packages-names (packageName: {
+                  ghcOptions = [
+                    "-ddump-to-file" "-ddump-hi"
+                    "-O0" "-Werror"
+                  ];
+                  postInstall = morley-infra.utils.${system}.weeder-hacks.collect-dump-hi-files;
+
+                  # enable haddock for local packages
+                  doHaddock = true;
+              });
+
+              # disable haddock for dependencies
+              doHaddock = false;
+            }
+            {
+              packages.stablecoin.components.library = {
+                preBuild = ''
```

Comment:
You can add `preHaddock` with the same contents and it should fix the necessity to do the same workaround with copying contracts for haddock build below